### PR TITLE
DOCSP-28586: How to use section

### PR DESCRIPTION
**Pull request by rustagir**
# Pull Request Info

[PR Reviewing Guidelines](https://github.com/mongodb/docs-java/blob/master/REVIEWING.md)

JIRA - https://jira.mongodb.org/browse/DOCSP-28586
Staging - https://docs-mongodbcom-staging.corp.mongodb.com/java/docsworker-xlarge/how-to-use-section/fundamentals/aggregation/aggregation-expression-operations/#how-to-use-operations

## Self-Review Checklist

- [x] Is this free of any warnings or errors in the RST?
- [x] Did you run a spell-check?
- [ ] Did you run a grammar-check?
- [x] Are all the links working?

### CODE REVIEW

File: source/fundamentals/aggregation/aggregation-expression-operations.txt [hunk 1]
```diff
@@ -40,45 +42,71 @@ guide, consider importing the following classes in your program:
...
+:ref:`Aggregates Builders <aggregates-builders>`.
+
+The examples use the ``Arrays.asList()`` method to create a list of
+aggregation stages that is passed to the ``aggregate()`` method of
+``MongoCollection``.
+
+Methods such as ``computed()`` and ``expr()`` that take values of type
+``TExpression`` allow you to use your expressions in aggregations. For
+example, the ``expr()`` method processes your operation as an expression
+that the ``match()`` method will accept.
```
Comment (rustagir) [inline source/fundamentals/aggregation/aggregation-expression-operations.txt:109]: I wonder if this should be in a note at line 96 instead.

File: source/fundamentals/aggregation/aggregation-expression-operations.txt [hunk 2]
```diff
@@ -40,45 +42,71 @@ guide, consider importing the following classes in your program:
...
-or ``Filters.expr()``. To reference document fields in an aggregation
-expression, obtain the "current" document being processed
-in the pipeline stage by calling the ``current()`` method. 
+To access document fields in an aggregation expression, you need to reference the
+document being processed by the aggregation pipeline. Use the
+``current()`` method to refer to this document.
 
-Since the aggregation operations are typed, you must specify the type of each
-field you reference using a typed variation of the ``getField()``
+To access the value of a field, you must use the appropriately typed
```
Comment (DBirtolo-mdb) [inline source/fundamentals/aggregation/aggregation-expression-operations.txt:49]: [suggestion]
I'd consider combining this with the paragraph above about using `current()` since they are closely tied together. It would also make the previous paragraph feel less like it is free floating.

File: source/fundamentals/aggregation/aggregation-expression-operations.txt [hunk 3]
```diff
@@ -40,45 +42,71 @@ guide, consider importing the following classes in your program:
...
-   While these new operations can be used within existing aggregation
-   methods that take ``TExpression`` values, methods that return
-   ``Bson`` values are not usable with this feature.
+- Checks if the size of the ``visitDates`` array is greater than ``0``
+  by using the ``gt()`` method 
+- Checks if the ``state`` field value is “New Mexico” by using the
+  ``eq()`` method
+
+The ``and()`` method links these operations so that the pipeline stage
+matches only documents that meet both criteria.
```
Comment (DBirtolo-mdb) [inline source/fundamentals/aggregation/aggregation-expression-operations.txt:80]: [suggestion]
Consider combining this above in the bulleted list. Something like

"Combines these two operations by using the `and()` method". 

If you do that, you might need to slightly reword the introductory sentence though.

File: source/fundamentals/aggregation/aggregation-expression-operations.txt [hunk 4]
```diff
@@ -40,45 +42,71 @@ guide, consider importing the following classes in your program:
...
+         .size()
+         .gt(of(0))
+         .and(current()
+                 .getString("state")
+                 .eq(of("New Mexico")));
+
+In most cases, you need to include your operation in a method such as
+``computed()`` or ``expr()`` to process it for use in an aggregation
+pipeline stage.
+
```
Comment (DBirtolo-mdb) [inline source/fundamentals/aggregation/aggregation-expression-operations.txt:96]: [question]
Do we want to explain at all why you might use computed or expr? Right now it is left as an exercise to the reader to figure out why one might want to use one or the other.

File: source/fundamentals/aggregation/aggregation-expression-operations.txt (hunk 1 above)
Comment (DBirtolo-mdb) [reply source/fundamentals/aggregation/aggregation-expression-operations.txt:109]: Considering my earlier comment, I would be inclined to agree. Putting this down here creates a disconnect from the knowledge. I don't know if it needs to be a tip, but I would definitely move it up.

**Review by DBirtolo-mdb — CHANGES_REQUESTED** [review]
Very nice work! This has been a beast and it's looking really good. Just a couple of suggestions to consider (not blocking) and one thing that I definitely think should be addressed.

File: source/fundamentals/aggregation/aggregation-expression-operations.txt (hunk 1 above)
Comment (rustagir) [reply source/fundamentals/aggregation/aggregation-expression-operations.txt:109]: made a note, LMK if you think the flow of info is good this way or if the content should be included as regular text.

File: source/fundamentals/aggregation/aggregation-expression-operations.txt (hunk 2 above)
Comment (rustagir) [reply source/fundamentals/aggregation/aggregation-expression-operations.txt:49]: agreed-- there are many short paragraphs in this section

File: source/fundamentals/aggregation/aggregation-expression-operations.txt (hunk 3 above)
Comment (rustagir) [reply source/fundamentals/aggregation/aggregation-expression-operations.txt:80]: I think the way it is now is a bit easier to follow, in the other case the last bullet point would reference the previous two, which I think is more challenging. For me, it's easier to understand this conceptually as two separate operations that are then linked

File: source/fundamentals/aggregation/aggregation-expression-operations.txt (hunk 4 above)
Comment (rustagir) [reply source/fundamentals/aggregation/aggregation-expression-operations.txt:96]: Chris, Maxim, and I met yesterday and spent a lot of time thinking/talking about this. We came to the conclusion that it might be going into the weeds too much to explain that a type conversion takes place, etc. etc. Easier just to say that users may need to use these methods to make the builders methods accept the expressions.

File: source/fundamentals/aggregation/aggregation-expression-operations.txt (hunk 1 above)
Comment (DBirtolo-mdb) [reply source/fundamentals/aggregation/aggregation-expression-operations.txt]: I think this works just fine as a note.

**Review by DBirtolo-mdb — APPROVED** [review]
LGTM

File: source/fundamentals/aggregation/aggregation-expression-operations.txt [hunk 5]
```diff
@@ -40,45 +42,72 @@ guide, consider importing the following classes in your program:
...
 
-.. note::
-   
-   While these new operations can be used within existing aggregation
-   methods that take ``TExpression`` values, methods that return
-   ``Bson`` values are not usable with this feature.
+To create a usable aggregation pipeline stage, include your expression
+in an aggregates builder method such as ``match(<expression>)`` or
+``project(<expression>)``. To learn more about these methods, see
+:ref:`Aggregates Builders <aggregates-builders>`.
```
Comment (katcharov) [inline source/fundamentals/aggregation/aggregation-expression-operations.txt:106]: The project and match methods do not take `<expression>`s directly. They accept plain BSON. That BSON can be in a format that is unrelated to expressions (like match's `field: 0 or 1` format). The `$expr` operator indicates that its contents are indeed an expression.

```suggestion
To create a usable aggregation pipeline stage, include your expression
in an aggregates builder method, such as in ``match(expr(<expression>))`` or
``project(fields(computed("<field name>", <expression>)))``. To learn more about these methods, see
:ref:`Aggregates Builders <aggregates-builders>`.
```

**Review by katcharov — CHANGES_REQUESTED** [review]
Looking great!

The only concerns are with the last part (stages and the computed/expr methods).

File: source/fundamentals/aggregation/aggregation-expression-operations.txt [hunk 6]
```diff
@@ -40,45 +42,72 @@ guide, consider importing the following classes in your program:
...
-method, such as ``getString()`` or ``getDate()``. The following code
-shows how to reference an integer field called ``quantity``:
+To specify a value in an operation, pass it to the ``of()`` method to
+convert it to a valid type. The following code shows how to reference a
+value of ``1.0``:
 
 .. code-block:: java
    :copyable: true
+   
+   of(1.0)
```
[the remark this answers is not in the document]
Comment (rustagir) [reply source/fundamentals/aggregation/aggregation-expression-operations.txt:64]: thanks for pointing that out! I think I'll have the chance to show `ofArray()` in the array operations section later on.

File: source/fundamentals/aggregation/aggregation-expression-operations.txt [hunk 7]
```diff
@@ -40,45 +42,72 @@ guide, consider importing the following classes in your program:
...
+         .getArray("visitDates")
+         .size()
+         .gt(of(0))
+         .and(current()
+                 .getString("state")
+                 .eq(of("New Mexico")));
+
+In most cases, you need to include your operation in a method such as
+``computed()`` or ``expr()`` to process it for use in an aggregation
+pipeline stage.
```
[the remark this answers is not in the document]
Comment (rustagir) [reply source/fundamentals/aggregation/aggregation-expression-operations.txt:94]: ok -- removing this para and taking the text out of the note!

[the remark this answers is not in the document]
Comment (rustagir) [reply source/fundamentals/aggregation/aggregation-expression-operations.txt:94]: adapted the text for flow

File: source/fundamentals/aggregation/aggregation-expression-operations.txt [hunk 8]
```diff
@@ -40,45 +42,72 @@ guide, consider importing the following classes in your program:
...
+.. note::
 
-When you finish composing an aggregation operation, you can pass it to
-an aggregation builder method, such as ``match()`` or ``project()`` to
-complete an aggregation pipeline stage. To learn more about these methods,
-see :ref:`Aggregates Builders <aggregates-builders>`.
+   Methods such as ``computed()`` and ``expr()`` that take values of
+   type ``TExpression`` allow you to use your expressions in
+   aggregations. For example, the ``expr()`` method processes your
+   operation as an expression that the ``match()`` method will accept.
```
[the remark this answers is not in the document]
Comment (rustagir) [reply source/fundamentals/aggregation/aggregation-expression-operations.txt:101]: disregard my reply on the previous PR comment -- how is this paragraph read now with revisions?

File: source/fundamentals/aggregation/aggregation-expression-operations.txt [hunk 9]
```diff
@@ -40,45 +42,75 @@ guide, consider importing the following classes in your program:
...
-The examples use the ``Arrays.asList()`` method to create a list of
-aggregation operations that can be passed as a pipeline to the
-``aggregate()`` method.
+To access document fields in an expression, you need to reference the
+document being processed by the aggregation pipeline. Use the
+``current()`` method to refer to this document. To access the value of a
+field, you must use the appropriately typed method, such as
+``getString()`` or ``getDate()``. When you specify the type for a field,
+you ensure that the driver recognizes its type correctly and lets you
+use methods that are compatible with that type. The following code
```
Comment (katcharov) [inline source/fundamentals/aggregation/aggregation-expression-operations.txt:51]:
```suggestion
you ensure that the driver provides only those
methods that are compatible with that type. The following code
```
(Avoiding the "recognition" (checking) and "letting" concepts since they are mildly misleading - for example, driver can't check types.)

File: source/fundamentals/aggregation/aggregation-expression-operations.txt (hunk 8 above)
[the remark this answers is not in the document]
Comment (katcharov) [reply source/fundamentals/aggregation/aggregation-expression-operations.txt]: Looks great!

File: source/fundamentals/aggregation/aggregation-expression-operations.txt [hunk 10]
```diff
@@ -40,45 +42,75 @@ guide, consider importing the following classes in your program:
...
+To complete your aggregation pipeline stage, include your expression
+in an aggregates builder method. The following list provides formats for
+how to include your expression in common aggregates builder methods:
+
+- ``match(expr(<expression>))``
+- ``project(fields(computed("<field name>", <expression>)))``
+- ``group(<expression>)``
+
+To learn more about these methods, see
+:ref:`Aggregates Builders <aggregates-builders>`.
```
[the remark this answers is not in the document]
Comment (rustagir) [reply source/fundamentals/aggregation/aggregation-expression-operations.txt:109]: the first sentence in the overview for the Aggregates Builders page (which is linked here) has a link to the API documentation!

**Review by katcharov — APPROVED** [review]
LGTM!
